Reorganize platformer example for readability

Koopa1018 opened the pull request:
The components and systems in the platformer example are confusingly sorted: components are stored in a separate file from the systems that use them, and within those files the components & systems are jumbled chaotically with little rhyme or reason. In a crate example this complex, this organization scheme makes understanding the code a real challenge (which is a major reason I couldn't update this example in #294).

This PR reorganizes the platformer example into modules sorted roughly by "part of the game" or "game system" (camera system, climbing, inventory, etc.). Each module contains both the components and ECS systems specific to that part, to present a complete picture of what the game system does and how it works.

No game logic has been changed. This PR focuses purely on code quality.

Koopa1018 committed:
Reorganize platformer example

THIS. EXAMPLE. IS. A MESS! How can anyone learn from a haphazard jumble like this? It's like grabbing an armful of notes from a physicist's desk and calling it a textbook....

Anyway. I need to be able to sort through this, so I just decided to go ape. No logic was changed--it's just organized intuitively now.

Koopa1018 committed:
Rustfmt reorganized platformer example

Trouv commented on examples/platformer/main.rs:
```diff
@@ -36,25 +44,17 @@ fn main() {
+        .add_systems(Update, camera::camera_fit_inside_current_level)
+        .add_plugins(misc_objects::MiscObjectsPlugin)
+        .register_ldtk_int_cell::<walls::WallBundle>(1)
+        .register_ldtk_int_cell::<climbing::LadderBundle>(2)
```
Could the `LadderBundle` be registered in the `LadderPlugin`?

Trouv reviewed (commented):
Great PR, much appreciated.

This really highlights how bloated and unfocused this example has become over the years. It's definitely overdue for some refactoring and this is great start.

I just have one non-NIT about the ladder registration, assuming CI passes

Trouv commented:
Clippy job should be fixed after #331

Koopa1018 committed:
Move wall system + cell registry to a WallPlugin

If we're consolidating related logic, it really does make sense to put all wall-related logic in a plugin.

Koopa1018 committed:
Move ladder int-cell reg to plugin also

Only there's an existing plugin for the climbing system.

Koopa1018 committed:
Move ground detection into own plugin+module

Koopa1018 committed:
Rename "physics" to "colliders"

Because that's all that's left after "ground_detection" is split off. (Also technically there's no physics calculations being done here.)

Koopa1018 commented:
Alright, changes complete.
